src/xsf/XSFFile.cc

```diff
@@ -0,0 +1,285 @@
+/*
```

Comment:
There is no license on this file.

Reply:
Oh, whoops. Thanks.

Reply:
Turns out it's because CyberBotX never formally put a license on it. I talked to him and got his permission to apply BSD to reflect the influence from AO.

I also forgot to put license headers on the files I pulled from my own project. Fixed that too.